### PR TITLE
feat(WAF): waf support global protection whitelist rule resource

### PR DESCRIPTION
**Pull request by deer-hang**
**What this PR does / why we need it**:
waf support global protection whitelist rule resource
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/waf' TESTARGS='-run TestAccRuleGlobalProtectionWhitelist_'
...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccRuleGlobalProtectionWhitelist_ -timeout 360m -parallel 4
=== RUN   TestAccRuleGlobalProtectionWhitelist_basic 
=== PAUSE TestAccRuleGlobalProtectionWhitelist_basic     
=== RUN   TestAccRuleGlobalProtectionWhitelist_withEpsID 
=== PAUSE TestAccRuleGlobalProtectionWhitelist_withEpsID 
=== CONT  TestAccRuleGlobalProtectionWhitelist_basic     
=== CONT  TestAccRuleGlobalProtectionWhitelist_withEpsID 
--- PASS: TestAccRuleGlobalProtectionWhitelist_withEpsID (401.27s) 
--- PASS: TestAccRuleGlobalProtectionWhitelist_basic (454.88s) 
PASS 
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       454.937s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/waf/resource_huaweicloud_waf_rule_global_protection_whitelist_test.go [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+    field    = "params"
+    logic    = "contain"
+    content  = "test content"
+    subfield = "test_subfield"
+  }
+
+  conditions {
+    field    = "ip"
+    logic    = "equal"
+    content  = "192.168.0.1"
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/waf/resource_huaweicloud_waf_rule_global_protection_whitelist_test.go]: remove the excess spaces

File: huaweicloud/services/acceptance/waf/resource_huaweicloud_waf_rule_global_protection_whitelist_test.go [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+    field    = "header"
+    logic    = "not_contain"
+    content  = "test header content"
+    subfield = "custom_subfield"
+  }
+
+  conditions {
+    field    = "url"
+    logic    = "prefix"
+    content  = "https://example.com"
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/waf/resource_huaweicloud_waf_rule_global_protection_whitelist_test.go]: ditto

**Comment by deer-hang** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/waf' TESTARGS='-run TestAccRuleGlobalProtectionWhitelist_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccRuleGlobalProtectionWhitelist_ -timeout 360m -parallel 4
=== RUN   TestAccRuleGlobalProtectionWhitelist_basic
=== PAUSE TestAccRuleGlobalProtectionWhitelist_basic
=== RUN   TestAccRuleGlobalProtectionWhitelist_withEpsID
=== PAUSE TestAccRuleGlobalProtectionWhitelist_withEpsID
=== CONT  TestAccRuleGlobalProtectionWhitelist_basic
=== CONT  TestAccRuleGlobalProtectionWhitelist_withEpsID
--- PASS: TestAccRuleGlobalProtectionWhitelist_withEpsID (387.05s) 
--- PASS: TestAccRuleGlobalProtectionWhitelist_basic (392.30s) 
PASS                                                           
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       392.371s
```

**Pull request merged**